crates/diesel_models/src/query/merchant_account.rs

```diff
@@ -123,4 +123,16 @@ impl MerchantAccount {
         )
         .await
     }
+
+    pub async fn update_all_merchant_accounts(
+        conn: &PgPooledConn,
+        merchant_account: MerchantAccountUpdateInternal,
+    ) -> StorageResult<Vec<Self>> {
+        generics::generic_update_with_results::<<Self as HasTable>::Table, _, _, _>(
+            conn,
+            dsl::merchant_id.ne_all(vec![""]),
+            merchant_account,
+        )
+        .await
```

Comment:
For this specific API, you could have used `generic_update()` which returns the number of rows updated. But I understand this function is meant to be generic enough.

Reply:
Used `update_with_results` to have a check if all the MerchantAccount returned from the list have desired value or not. But left it because seemed redundant